Source/Core/Common/OnionConfig.cpp

```diff
@@ -0,0 +1,380 @@
+		{
+			std::string temp;
+			bool retval = Get(key, &temp);
+			if (!retval || temp.empty())
+			{
+				return false;
+			}
+
+			// ignore starting comma, if any
+			size_t subStart = temp.find_first_not_of(",");
+
+			// split by comma
+			while (subStart != std::string::npos)
+			{
+				// Find next comma
+				size_t subEnd = temp.find(',', subStart);
+				if (subStart != subEnd)
+				{
+					// take from first char until next comma
+					out->push_back(StripSpaces(temp.substr(subStart, subEnd - subStart)));
+				}
+
+				// Find the next non-comma char
+				subStart = temp.find_first_not_of(",", subEnd);
+			}
+
+			return true;
+		}
+
+		// Onion layers
+		BloomLayer::BloomLayer(std::unique_ptr<ConfigLayerLoader> loader)
+			: m_layer(loader->GetLayer()), m_loader(std::move(loader))
+		{
+			Load();
+		}
+
+		BloomLayer::~BloomLayer()
+		{
+			Save();
+		}
+
+		OnionPetal* BloomLayer::GetPetal(OnionSystem system, const std::string& petal_name)
+		{
+			for (OnionPetal* petal : m_petals[system])
+				if (!strcasecmp(petal->m_name.c_str(), petal_name.c_str()))
+					return petal;
+			return nullptr;
+		}
+
+		OnionPetal* BloomLayer::GetOrCreatePetal(OnionSystem system, const std::string& petal_name)
+		{
+			OnionPetal* petal = GetPetal(system, petal_name);
+			if (!petal)
+			{
+				if (m_layer == OnionLayerType::LAYER_META)
+					m_petals[system].emplace_back(new RecursiveOnionPetal(m_layer, system, petal_name));
+				else
+					m_petals[system].emplace_back(new OnionPetal(m_layer, system, petal_name));
+				petal = m_petals[system].back();
+			}
+			return petal;
+		}
+
+		void BloomLayer::Load()
+		{
+			if (m_loader)
+				m_loader->Load(this);
+		}
+
+		void BloomLayer::Save()
+		{
+			if (m_loader)
+				m_loader->Save(this);
+		}
+
+	OnionPetal* GetOrCreatePetal(OnionSystem system, const std::string& petal_name)
+	{
+		return m_layers[OnionLayerType::LAYER_META]->GetOrCreatePetal(system, petal_name);
+	}
+
+	OnionBloom* GetFullBloom()
+	{
+		return &m_layers;
+	}
+
+	void AddLayer(BloomLayer* layer)
+	{
+		m_layers[layer->GetLayer()] = layer;
+	}
+
+	void AddLayer(ConfigLayerLoader* loader)
+	{
+		BloomLayer* layer = new BloomLayer(std::unique_ptr<ConfigLayerLoader>(loader));
+		m_layers[layer->GetLayer()] = layer;
+	}
+
+	void RemoveLayer(OnionLayerType layer)
+	{
+		m_layers.erase(layer);
+	}
+	bool LayerExists(OnionLayerType layer)
+	{
+		return m_layers.find(layer) != m_layers.end();
+	}
+
+	// Explicit load and save of layers
+	void Load()
+	{
+		for (auto& layer : m_layers)
+			layer.second->Load();
+	}
+
+	void Save()
+	{
+		for (auto& layer : m_layers)
+			layer.second->Save();
+	}
+
+	void Init()
+	{
+		// This layer always has to exist
+		m_layers[OnionLayerType::LAYER_META] = new RecursiveBloomLayer();
+	}
+
+	void Shutdown()
+	{
+		auto it = m_layers.begin();
+		while (it != m_layers.end())
+		{
+			delete it->second;
+			it = m_layers.erase(it);
+		}
+	}
+
+	std::string& GetSystemName(OnionSystem system)
+	{
```

Comment:
If it doesn't intend to be modified, it should be a ```const std::string&``` and the ```static std::map``` should be ```const``` as well. Then use the ```at()``` method to access the value instead of the ```operator[]```.